samcli/commands/sync/core/command.py

```diff
@@ -0,0 +1,141 @@
+"""
+Sync Command Class.
+"""
+from click import Command, Context, Parameter, style
+
+from samcli.cli.row_modifiers import RowDefinition, ShowcaseRowModifier
+from samcli.commands.sync.core.formatters import SyncCommandHelpTextFormatter
+from samcli.commands.sync.core.options import OPTIONS_INFO
+
+SYNC_DESCRIPTION = """
+  By default, `$sam sync` runs a full AWS Cloudformation stack update. 
+
+  Running `sam sync --watch` with `--code` will provide a way to run just code
+  synchronization, speeding up start time skipping template changes.
+  
+  Remember to update the deployed stack by running
+  without --code for infrastructure changes.
+  
+  `$sam sync` also supports nested stacks and nested stack resources.
+  
+  """ + style(
+    "This command requires access to AWS credentials.", bold=True
+)
```

Comment:
Just a thought, but creating an abstraction around descriptions that we can pass to the command decorator that takes a description string and a boolean `requires_credentials` that it can format.

Reply:
yeah, good callout. I'll think about a refactor.